### PR TITLE
Adding blog on custom detection rules in Security Analytics

### PR DESCRIPTION
**Pull request by xeniatup**
### Description
Adding blog content for an editorial review and publishing
 
### Issues Resolved
https://github.com/opensearch-project/project-website/issues/2383

### Check List
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the BSD-3-Clause License.

### CODE REVIEW

File: _posts/2023-11-28-how-to-create-custom-threat-detection-rules.md [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+
+You can add multiple **Selections** with multiple Maps for the log fields. **Map** is a directory that contains key-value pairs. **Modifiers** transform values and lists or convert them into regular expressions. To learn more about the Sigma rule format, go to [documentation](https://github.com/SigmaHQ/sigma). 
+
+- Create a Map with **Key**: `UserID`, **Modifier**: `contains`, and **Value**: `<your organization domain name>`.
+- Create a Map with **Key**: `Recipient`, **Modifier**: `endswith`, and **List**: `<values that include the domains to be detected>`. 
+
+![Selection_1 section](/assets/media/blog-images/2023-11-28-how-to-create-custom-threat-detection-rules/selection1.png)
+
+6. Create another **Selection** where **Key**: `UserId` is a list of emails belonging to the marketing team by uploading the list from a .csv (or .txt) file. 
+
```
Comment (vagimeli) [inline _posts/2023-11-28-how-to-create-custom-threat-detection-rules.md:76]: Delete extra blank line

File: _posts/2023-11-28-how-to-create-custom-threat-detection-rules.md [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+
+![Upload a file dialog](/assets/media/blog-images/2023-11-28-how-to-create-custom-threat-detection-rules/upload-a-file.png)
+
+After uploading the list, **Selection_2** will look like this:
+
+![Selection_2 section](/assets/media/blog-images/2023-11-28-how-to-create-custom-threat-detection-rules/selection2.png)
+
+7. Enter  `Selection_1 AND NOT include Selection_2` in the **Condition** fields to define how each **Selection** will contribute to the final query .
```
Comment (vagimeli) [inline _posts/2023-11-28-how-to-create-custom-threat-detection-rules.md:84]:
```suggestion
7. Enter  `Selection_1 AND NOT include Selection_2` in the **Condition** fields to define how each **Selection** will contribute to the final query. 
```

File: _posts/2023-11-28-how-to-create-custom-threat-detection-rules.md [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+![Upload a file dialog](/assets/media/blog-images/2023-11-28-how-to-create-custom-threat-detection-rules/upload-a-file.png)
+
+After uploading the list, **Selection_2** will look like this:
+
+![Selection_2 section](/assets/media/blog-images/2023-11-28-how-to-create-custom-threat-detection-rules/selection2.png)
+
+7. Enter  `Selection_1 AND NOT include Selection_2` in the **Condition** fields to define how each **Selection** will contribute to the final query . 
+
+The UI allows the combining of multiple selections using operators, including `AND`, `AND NOT`, `OR`, and `OR NOT`.
```
Comment (vagimeli) [inline _posts/2023-11-28-how-to-create-custom-threat-detection-rules.md:86]:
```suggestion
The user interface (UI) allows for the combining of multiple selections using operators, including `AND`, `AND NOT`, `OR`, and `OR NOT`. 
```

File: _posts/2023-11-28-how-to-create-custom-threat-detection-rules.md [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+
+After uploading the list, **Selection_2** will look like this:
+
+![Selection_2 section](/assets/media/blog-images/2023-11-28-how-to-create-custom-threat-detection-rules/selection2.png)
+
+7. Enter  `Selection_1 AND NOT include Selection_2` in the **Condition** fields to define how each **Selection** will contribute to the final query . 
+
+The UI allows the combining of multiple selections using operators, including `AND`, `AND NOT`, `OR`, and `OR NOT`. 
+
+**Pro tip**: If you’re more comfortable using YAML formats, you can switch to the YAML editor on the top of the page under the page title at any point and write a **Condition** as code without losing the context.
```
Comment (vagimeli) [inline _posts/2023-11-28-how-to-create-custom-threat-detection-rules.md:88]:
```suggestion
**Pro tip**: If you’re more comfortable using YAML formats, you can switch to the YAML editor at the top of the page under the page title at any point and write a **Condition** as code without losing the context.
```

File: _posts/2023-11-28-how-to-create-custom-threat-detection-rules.md [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+
+![Condition section](/assets/media/blog-images/2023-11-28-how-to-create-custom-threat-detection-rules/condition.png) 
+
+Once a custom rule is added, it will show up on the **Detection rules** page and can be used in any new or existing detector for customized fine-grained threat detection.
+
+### What are the benefits?
+
+A custom approach to detection rules that are tailored to your unique business cases helps target specific malicious activity across a range of cloud environments and troubleshoot application-specific threats. OpenSearch Security Analytics provides the capability to configure alerts and customize notification messages, helping users to find threats that need further investigation. By using the custom detection rules workflow, you can expand the detection criteria to include custom application log sources and monitor them for specific threat patterns. You can also use correlation rules to find patterns between standard and custom application logs.
```
Comment (vagimeli) [inline _posts/2023-11-28-how-to-create-custom-threat-detection-rules.md:96]:
```suggestion
A custom approach to detection rules that are tailored to your unique business cases helps target specific malicious activity across a range of cloud environments and troubleshoot application-specific threats. OpenSearch Security Analytics provides the capability to configure alerts and customize notification messages, helping you to find threats that need further investigation. By using the custom detection rules workflow, you can expand the detection criteria to include custom application log sources and monitor them for specific threat patterns. You can also use correlation rules to find patterns between standard and custom application logs. 
```

File: _posts/2023-11-28-how-to-create-custom-threat-detection-rules.md [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+### What are the benefits?
+
+A custom approach to detection rules that are tailored to your unique business cases helps target specific malicious activity across a range of cloud environments and troubleshoot application-specific threats. OpenSearch Security Analytics provides the capability to configure alerts and customize notification messages, helping users to find threats that need further investigation. By using the custom detection rules workflow, you can expand the detection criteria to include custom application log sources and monitor them for specific threat patterns. You can also use correlation rules to find patterns between standard and custom application logs. 
+
+Integrating custom rules to the threat detection strategies empowers your security teams to enhance coverage for detecting security threats across a variety of log sources. Custom rules can be written with a deep understanding of an organization's infrastructure, making them more precise in identifying specific threats and reducing false positives. Combining custom and pre-packaged rules for threat detection strikes a balance between tailored and standardized protection.
```
Comment (vagimeli) [inline _posts/2023-11-28-how-to-create-custom-threat-detection-rules.md:98]:
```suggestion
Integrating custom rules into the threat detection strategies empowers your security teams to enhance coverage for detecting security threats across a variety of log sources. Custom rules can be written with a deep understanding of an organization's infrastructure, making them more precise in identifying specific threats and reducing false positives. Combining custom and prepackaged rules for threat detection strikes a balance between tailored and standardized protection.
```

File: _posts/2023-11-28-how-to-create-custom-threat-detection-rules.md [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+
+To get started on creating a custom detection rule for a custom log source or application log, refer to [documentation](https://opensearch.org/docs/latest/security-analytics/usage/rules/#creating-detection-rules). To learn more about security analytics and its capabilities, see the [documentation](https://opensearch.org/docs/latest/security-analytics/index/).
```
Comment (vagimeli) [inline _posts/2023-11-28-how-to-create-custom-threat-detection-rules.md:100]:
```suggestion
To get started on creating a custom detection rule for a custom log source or application log, refer to [documentation](https://opensearch.org/docs/latest/security-analytics/usage/rules/#creating-detection-rules). To learn more about security analytics and its capabilities, see the [Working with detection rules documentation](https://opensearch.org/docs/latest/security-analytics/index/).
```

**Review by vagimeli — APPROVED** [review]
@xeniatup Outstanding job on the writing. It's clear, concise, and reader-friendly. Editorial review is complete. Edits are primarily copyedits for consistent usage. Please reach out if needed. Thanks!

**Comment by xeniatup** [comment]
@vagimeli Thanks for the review! I committed all the changes.

File: _posts/2023-11-28-how-to-create-custom-threat-detection-rules.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+title:  "Building custom threat detection rules with OpenSearch Security Analytics"
+authors:
+- xeniatup
+- jimishs
+- sandeshkr419
+date: 2023-11-28
+categories:
+ - feature
+ - technical-posts
+meta_keywords: threat detection, security analytics, SIEM, detection rules
```
Comment (pajuric) [inline _posts/2023-11-28-how-to-create-custom-threat-detection-rules.md:12]: Please update the following: 

**meta_keywords:** OpenSearch security analytics, threat detection, security analytics plugin, detection rules

**Comment by pajuric** [comment]
@nateynateynate @krisfreedain @dtaivpp - Blog is ready to publish on Tuesday, November 21.

**Pull request merged**